x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/TimeSeriesAggregationOperatorTests.java

```diff
@@ -177,229 +203,114 @@ record Doc(String pod, long timestamp, long requests) {
+            List.of(new RateLongAggregatorFunctionSupplier(List.of(2, 3, 4), unitInMillis)),
+            List.of(),
+            between(1, 100)
         ).get(ctx);
+        // tsid, bucket, rate, grouping1, grouping2
+        List<BlockHash.GroupSpec> finalGroups = new ArrayList<>();
+        int groupChannel = 3;
+        for (String grouping : groupings) {
+            if (grouping.equals("bucket")) {
+                finalGroups.add(new BlockHash.GroupSpec(1, ElementType.LONG));
+            } else {
+                finalGroups.add(new BlockHash.GroupSpec(groupChannel++, ElementType.BYTES_REF));
+            }
+        }
+        Operator finalAgg = new TimeSeriesAggregationOperatorFactories.Final(
+            finalGroups,
+            List.of(new SumDoubleAggregatorFunctionSupplier(List.of(2))),
+            List.of(),
+            between(1, 100)
+        ).get(ctx);
+
         List<Page> results = new ArrayList<>();
-        var requestsField = new NumberFieldMapper.NumberFieldType("requests", NumberFieldMapper.NumberType.LONG);
         OperatorTestCase.runDriver(
             new Driver(
                 ctx,
                 sourceOperatorFactory.get(ctx),
-                List.of(ValuesSourceReaderOperatorTests.factory(reader, requestsField, ElementType.LONG).get(ctx), initialHash, finalHash),
+                CollectionUtils.concatLists(extractOperators, List.of(intialAgg, intermediateAgg, finalAgg)),
                 new TestResultPageSinkOperator(results::add),
                 () -> {}
             )
         );
-        Map<Group, Double> rates = new HashMap<>();
+        List<List<Object>> values = new ArrayList<>();
         for (Page result : results) {
-            BytesRefBlock keysBlock = result.getBlock(0);
-            LongBlock timestampIntervalsBock = result.getBlock(1);
-            DoubleBlock ratesBlock = result.getBlock(2);
-            for (int i = 0; i < result.getPositionCount(); i++) {
-                var key = new Group(keysBlock.getBytesRef(i, new BytesRef()).utf8ToString(), timestampIntervalsBock.getLong(i));
-                rates.put(key, ratesBlock.getDouble(i));
+            for (int p = 0; p < result.getPositionCount(); p++) {
+                int blockCount = result.getBlockCount();
+                List<Object> row = new ArrayList<>();
+                for (int b = 0; b < blockCount; b++) {
+                    row.add(BlockUtils.toJavaObject(result.getBlock(b), p));
+                }
+                values.add(row);
             }
             result.releaseBlocks();
         }
-        return rates;
-    }
-
-    record Group(String tsidHash, long timestampInterval) {}
-
-    // TODO: in a follow up add support for ordinal based time series grouping operator
-    // (and then remove this test)
-    // (ordinal based can only group by one field and never includes timestamp)
-    public void testBasicRateOrdinalBased() {
```

Comment:
I am removing this as we won't run the rate aggregations with the ordinals grouping.